ci/cleanup-pr-previews

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# A utility script to clean up PR preview documentation folders for closed/merged PRs.
+# This script checks all pr-XXXXX folders in the gh-pages branch docs/pr-preview/ directory,
+# verifies if the corresponding PR XXXXX is still open, and removes preview folders
+# for PRs that have been closed or merged.
+
+set -euo pipefail
+
+# Configuration
+REPOSITORY="${1:-NVIDIA/cuda-python}"
```

Comment:
I don't want this to be configurable. There are two problems:
1. If I want to do dry run (which is almost always required), I am forced to also specify the repo name which is annoying
2. This script is tailored for the structure of this particular repo, not generalizable easily to arbitrary repos, and I don't want to give that impression